pxr/usd/usd/stage.cpp

```diff
@@ -9110,6 +9110,30 @@ UsdStage::SetFramesPerSecond(double framesPerSecond) const
     SetMetadata(SdfFieldKeys->FramesPerSecond, framesPerSecond);
 }
 
+std::vector<double>
+UsdStage::GetTimeSamplesFromPrims()
```

Comment:
Some notes on this method:
1. The main argument for having this method here is to keep it near the other `_GetTimeSamples*()` family of methods. But we could also put it somewhere else, maybe as a utility, since it's use case might not be fully generic and its cost is a bit high due to the stage traversal. However, accessing it from the stage is pretty convenient.
2. Currently this doesn't take any arguments, since that wasn't needed for our feature. It may make sense to provide a few more options if this method stays here... For example, we could add an option for a `Usd_PrimFlagsPredicate` to guide the stage traversal.